eclipse-collections-code-generator/src/main/resources/impl/bag/immutable/immutablePrimitiveSingletonBag.stg

```diff
@@ -214,9 +214,10 @@ final class Immutable<name>SingletonBag implements Immutable<name>Bag, Serializa
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public \<V> ImmutableBag\<V> collect(<name>ToObjectFunction\<? extends V> function)
     {
-        return HashBag.newBagWith(function.valueOf(this.element1)).toImmutable();
+        return (ImmutableBag\<V>) HashBag.newBagWith(function.valueOf(this.element1)).toImmutable();
```

Comment:
This could be simplified with:

`return Bags.immutable.with(function.valueOf(this.element1));
`

This addresses a compiler error in IntelliJ so perhaps it will work in Eclipse as well.

This would require an import as well:

`import org.eclipse.collections.impl.factory.Bags;
`